build.sbt

```diff
@@ -332,43 +301,64 @@ lazy val docs = http4sProject("docs")
+    makeSite := makeSite.dependsOn(tutQuick, exportMetadataForSite).value,
     baseURL in Hugo := {
-      if (isTravisBuild.value) new URI(s"http://http4s.org")
+      if (isTravisBuild.value) new URI(s"http://http4s.org/v0.15")
       else new URI(s"http://127.0.0.1:${previewFixedPort.value.getOrElse(4000)}")
     },
-    // all .md|markdown files go into `content` dir for hugo processing
-    ghpagesNoJekyll := true,
     includeFilter in Hugo := (
         "*.html" |
         "*.png" | "*.jpg" | "*.gif" | "*.ico" | "*.svg" |
         "*.js" | "*.swf" | "*.json" | "*.md" |
         "*.css" | "*.woff" | "*.woff2" | "*.ttf" |
         "CNAME" | "_config.yml"
     ),
-    siteMappings := {
-      if (Http4sGhPages.buildMainSite) siteMappings.value
-      else {
-        val (major, minor) = apiVersion.value
-        val prefix = s"/v${major}.${minor}/"
-        siteMappings.value.filter {
-          case (_, d) if d.startsWith(prefix) => true
-          case _ => false
-        }
-      }
-    },
     siteMappings ++= {
       val m = (mappings in (ScalaUnidoc, packageDoc)).value
       val (major, minor) = apiVersion.value
-      for ((f, d) <- m) yield (f, s"v$major.$minor/api/$d")
+      for ((f, d) <- m) yield (f, s"api/$d")
+    },
+    includeFilter in ghpagesCleanSite := {
+      new FileFilter{
+        def accept(f: File) =
+          f.getCanonicalPath.startsWith((ghpagesRepository.value / "v0.15").getCanonicalPath)
+      }
+    },
+    ghpagesPushSite ~= { old =>
+      if (sys.env.get("TRAVIS_BRANCH") == Some("master")) old
+      else ()
     },
-    cleanSite := Http4sGhPages.cleanSiteForRealz(updatedRepository.value, gitRunner.value, streams.value, apiVersion.value),
-    synchLocal := Http4sGhPages.synchLocalForRealz(privateMappings.value, updatedRepository.value, ghpagesNoJekyll.value, gitRunner.value, streams.value, apiVersion.value),
     git.remoteRepo := "git@github.com:http4s/http4s.git"
   )
   .dependsOn(client, core, theDsl, blazeServer, blazeClient, circe)
 
+lazy val website = http4sProject("website")
+  .settings(noPublishSettings)
+  .settings(noCoverageSettings)
+  .enablePlugins(HugoPlugin, GhpagesPlugin)
+  .settings(
+    description := "Common area of http4s.org",
+    baseURL in Hugo := {
+      if (isTravisBuild.value) new URI(s"http://http4s.org")
+      else new URI(s"http://127.0.0.1:${previewFixedPort.value.getOrElse(4000)}")
+    },
+    // all .md|markdown files go into `content` dir for hugo processing
+    ghpagesNoJekyll := true,
+    includeFilter in Hugo := (
+        "*.html" |
+        "*.png" | "*.jpg" | "*.gif" | "*.ico" | "*.svg" |
+        "*.js" | "*.swf" | "*.json" | "*.md" |
+        "*.css" | "*.woff" | "*.woff2" | "*.ttf" |
+        "CNAME" | "_config.yml"
+    ),
+    excludeFilter in ghpagesCleanSite := {
+      new FileFilter{
+        def accept(f: File) =
+          f.getCanonicalPath.startsWith((ghpagesRepository.value / "v0.*").getCanonicalPath)
```

Comment:
This is going to bite us when we go 1.0.  I don't know how to cleanly target the `/^v[0-9]+\./` directories.  An alternative would be to put all the tutorials in a `docs/` subdirectory of the published site (as is common on other projects), but that would change all our URIs.